CMakeLists.txt

```diff
@@ -0,0 +1,102 @@
+cmake_minimum_required(VERSION 3.4)
```

Comment:
Is this truly the minimum required version? I removed this line and ran cmake 2.8.12.2 (the default version on CentOS 7) and everything seemed to build properly. I'm leery of requiring versions that aren't installed by default on relatively modern OS releases.

Reply:
You are correct. This is the latest version I have ran the script with. I could not find a tool to compute the minimum required version and browsing the documentaiton seamed to be an error-prone effort. 

I did some manual testing with older version of CMake and fixing the if-clause as you proposed makes it is possible to run CMake 2.4.4 successfully. CMake 2.4.2 fail to parse the EXCLUDE_FROM_ALL option and I can not find the documentation for this version? The option is defined in the CMake 2.4.4 documenation at least: https://cmake.org/Wiki/CMake_2.4.4_Docs

If you can find some earlier use of this option we could go for an older CMake version. Otherwise we could have a look at some popular package repositories to try to determine the oldest CMake version still in use and make this the minimum required version?

Reply:
2.4.4 seems plenty old. Let's go with that.